lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
```

Comment:
`match?({0, _}, :binary.match(info, ["<<<", "===", ">>>", ...]))` will match only on the front. You can also do normal binary pattern matching:

```elixir
defp match?("<<<" <> _), do: true
defp match?("===" <> _), do: true
defp match?(_), do: false
```

It's a good idea to avoid regular expressions when possible.